hack/update-mocks.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates mock files using mockgen.
+# Usage: `hack/update-mocks.sh`.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+
+_tmp="${KUBE_ROOT}/_tmp_build_tag_files"
+mkdir -p "${_tmp}"
+
+function cleanup {
+	rm -rf "$_tmp"
+	rm -f "tempfile"
+}
+
+trap cleanup EXIT
+
+kube::util::ensure_clean_working_dir
+kube::golang::verify_go_version
+
+echo 'installing mockgen'
+pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+  go install github.com/golang/mock/mockgen
+popd >/dev/null
+
+find_files() {
+  find . -not \( \
+      \( \
+        -wholename './output' \
+        -o -wholename './.git' \
+        -o -wholename './_output' \
+        -o -wholename './_gopath' \
+        -o -wholename './release' \
+        -o -wholename './target' \
+        -o -wholename '*/third_party/*' \
+        -o -wholename '*/vendor/*' \
+        -o -wholename './staging/src/k8s.io/client-go/*vendor/*' \
+        -o -wholename '*/bindata.go' \
+      \) -prune \
+    \) -name '*.go'
+}
+
+cd "${KUBE_ROOT}"
+
+echo 'executing go generate command on below files'
+
+for IFILE in $(find_files | xargs git grep --files-with-matches -e '//go:generate mockgen'); do
```

Comment:
```suggestion
for IFILE in $(find_files | xargs grep --files-with-matches -e '//go:generate mockgen'); do
```

Should we expand the scope of update-mocks to also include files that are tracked by git yet? For example, if I add something in `pkg/foo/foo.go` that I want to mock. I need to add it to the index, commit and then run the script.

Reply:
agreed!

used `grep` instead of `git grep` so that other non tracked files can be considered